### ISSUE TITLE
When Optikey is running applications in taskbar open a new instance instead of maximizing.

### ISSUE THREAD

**Issue by SamPlusPlus** [JuliusSweetland/OptiKey#102]
I apologize in advance, this issue seems a little weird to me, and I am not sure if the behavior was intended by design.

To reproduce: 

Without Optikey open.

Open calc.exe and minimize it to the taskbar. Double click on  the application icon running in the taskbar. The calculator window is restored, and if done again is minimized to the taskbar.

Now with Optikey open, run the same test, this time you should notice a new instance of calculator is launched, and the running process continues to run, resulting in two instances of calc.exe

This is a simple example, but this issue has some strange side effect, like selecting Optikey while its running, you can end up with two on screen keyboards.

Can you elaborate on this behavior, is this by design, or is this a bug. If by design what is the intended use case?

As a slightly related issue, though we can move to a new ticket if needed. Should Optikey allow multiple instances of the application to run at one time?

**Comment by JuliusSweetland** [comment]
This is due to the fact that OptiKey has the shift key pressed down (possibly due to auto capitalisation). Try without OptiKey running;

open a calculator and minimise
hold down shift
click on the calculator icon in the taskbar to restore the calculator
instead you will get a second instance of calculator

This behavior is dealt with when using OptiKey to simulate mouse actions as modifier keys are released when the user is within the mouse keyboard.

Closing issue as it is by design and only a problem when mixing OptiKey and the use of the physical mouse, which isn't necessarily a real use case.

**Comment by SamPlusPlus** [comment]
I do not have an eye tracker to test with, but would this not be an issue when switching between applications such as an instance of Word and Chrome that are already open?

Secondly you mention Optikey is using the shiftkey for auto capitalization, would it be possible to use the Caps Lock key instead and avoid the undesired behavior?

**Comment by JuliusSweetland** [comment]
@SamPlusPlus I get around this being a problem by temporarily disabling all modifier keys when the user switches to the mouse screen, and then reinstating everything when they leave the mouse screen. It works pretty well and this behaviour can also be disabled should the user want to do something complex with their modifier keys + mouse actions.

I thought about using CAPS, and might return to this, but the SHIFT key behaves more like a normal UP/DOWN/HELD DOWN key, whereas the CAPS key is toggled on/off by pressing and releasing, which isn't really supported in my UI at the moment. I will consider it, but if OptiKey is controlled fully using an eye tracker it's not really a problem. I've spent quite a lot of time thinking about how to mitigate the problems with OptiKey pressing (especially auto-pressing) modifier keys. I'm reasonably happy with where I've got to, but I'm always mulling these problems over, so I might change it all at some point!

Thanks for your feedback.
